airflow/models/dag.py

```diff
@@ -1347,8 +1347,18 @@ def __deepcopy__(self, memo):
             result._log = self._log
         return result
 
-    def sub_dag(self, task_regex, include_downstream=False,
-                include_upstream=True):
+    def sub_dag(self, *args, **kwargs):
+        """This method is deprecated in favor of partial_subset"""
+        warnings.warn(
+            "This method is deprecated and will be removed in a future version. Please use partial_subset",
+            DeprecationWarning,
+            stacklevel=2,
```

Comment:
Can we add this in Updating.md too, please ?

Reply:
I kind of didn't on purpose - it's so rare to be used, and anyone who does will get a Depreciation warning

Reply:
👍 Yea cool